packages/eslint-plugin/package.json

```
@@ -102,7 +102,7 @@
     "unist-util-visit": "^5.0.0"
   },
   "peerDependencies": {
-    "@typescript-eslint/parser": "^7.0.0",
+    "@typescript-eslint/parser": "^8.0.0",
```

Note that `^` will not match "pre release" ranges by default.
If you add the base, re-release version then semver will match all those pre-release versions

```suggestion
    "@typescript-eslint/parser": "^8.0.0 || ^8.0.0-alpha.0",
```